### PR TITLE
Doing lesson 17

### PR DESCRIPTION
**Pull request by TatsianaKazachenka**

### CODE REVIEW

File: src/main/java/pages/FeedPage.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        $(UPDATE_MOOD_BTN_CSS).shouldBe(Condition.visible, Duration.ofSeconds(10));
+        return this;
+    }
+
+    public RateYourHappiness clickOnUpdateBtn() {
+        $(UPDATE_MOOD_BTN_CSS).click();
+        return new RateYourHappiness();
+    }
+
+    public String textPost() {
```
Comment (vchesnakoutest) [inline src/main/java/pages/FeedPage.java]: getPostText()

File: src/main/java/pages/FeedPage.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        String text = $(FIRST_POST_CSS).find(TEXT_POST_CSS).getText();
+        return text;
+    }
+
+    public int moodLevel() {
```
Comment (vchesnakoutest) [inline src/main/java/pages/FeedPage.java]: getMoodLevel()

File: src/main/java/pages/RateYourHappiness.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package pages;
+
+import org.openqa.selenium.Keys;
+
+import static com.codeborne.selenide.Selenide.$;
+
+public class RateYourHappiness {
```
Comment (vchesnakoutest) [inline src/main/java/pages/RateYourHappiness.java]: RateYourHappinessPage

File: src/main/java/steps/MoodUpdateModalSteps.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import pages.*;
+
+public class MoodUpdateModalSteps {
+    private FeedPage page;
+
+    public MoodUpdateModalSteps() {
+        page = new FeedPage();
+    }
+
+    public MoodUpdateModal fullingFieldAndGoToDiary(int moodValue, String description) {
```
Comment (vchesnakoutest) [inline src/main/java/steps/MoodUpdateModalSteps.java]: updateMoodAndGoToDiary

File: src/test/java/tests/MoodPandaTest.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+public class MoodPandaTest extends BaseTest {
+
+    @Test(description = "check the value update on the button")
+    public void updateMoodValueTest() {
+        int moodValue = 8;
+        String description = "Hello!";
+        loginSteps.login(EMAIL, PASSWORD)
+                .clickOnUpdateBtn()
+                .updateMood(moodValue, description);
+        int getMoodValue = rateYourHappiness.getMoodValueHappiness();
```
Comment (vchesnakoutest) [inline src/test/java/tests/MoodPandaTest.java:15]: int moodValue (use 'get' in methods names)

File: src/test/java/tests/MoodPandaTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        Assert.assertEquals(getMoodValue, moodValue);
+    }
+
+    @Test(description = "check the description update in the post")
+    public void DescriptionTest() {
```
Comment (vchesnakoutest) [inline src/test/java/tests/MoodPandaTest.java]: methods should be named in camelCase: descriptionTest

File: src/test/java/tests/MoodPandaTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        int moodValue = 8;
+        String description = "Hello!";
+        loginSteps.login(EMAIL, PASSWORD);
+        moodUpdateModalSteps.fullingFieldAndGoToDiary(moodValue, description);
+        String textPost = feedPage.textPost();
+        Assert.assertEquals(textPost, description);
+    }
+
+    @Test(description = "check the value mood update in the post")
+    public void LevelMoodTest() {
```
Comment (vchesnakoutest) [inline src/test/java/tests/MoodPandaTest.java]: methods should be named in camelCase: levelMoodTest

File: src/test/java/tests/MoodPandaTest.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        int moodValue = 8;
+        String description = "Hello!";
+        loginSteps.login(EMAIL, PASSWORD);
+        moodUpdateModalSteps.fullingFieldAndGoToDiary(moodValue, description);
+        int level = feedPage.moodLevel();
+        Assert.assertEquals(moodValue, level);
+    }
+
+    @Test(description = "check the number of hugs after clicking")
```
Comment (vchesnakoutest) [inline src/test/java/tests/MoodPandaTest.java:39]: methods should be named in camelCase: hugTest

**Review by vchesnakoutest — CHANGES_REQUESTED** [review]
make fixes and merge to master

File: src/main/java/pages/FeedPage.java (hunk 1 above)
Comment (TatsianaKazachenka) [reply src/main/java/pages/FeedPage.java]: done

File: src/main/java/pages/FeedPage.java (hunk 2 above)
Comment (TatsianaKazachenka) [reply src/main/java/pages/FeedPage.java]: done

File: src/main/java/pages/RateYourHappiness.java (hunk 3 above)
Comment (TatsianaKazachenka) [reply src/main/java/pages/RateYourHappiness.java]: done

File: src/main/java/steps/MoodUpdateModalSteps.java (hunk 4 above)
Comment (TatsianaKazachenka) [reply src/main/java/steps/MoodUpdateModalSteps.java]: done

File: src/test/java/tests/MoodPandaTest.java (hunk 6 above)
Comment (TatsianaKazachenka) [reply src/test/java/tests/MoodPandaTest.java]: done

File: src/test/java/tests/MoodPandaTest.java (hunk 7 above)
Comment (TatsianaKazachenka) [reply src/test/java/tests/MoodPandaTest.java]: done

File: src/test/java/tests/MoodPandaTest.java (hunk 8 above)
Comment (TatsianaKazachenka) [reply src/test/java/tests/MoodPandaTest.java:39]: done

File: src/test/java/tests/MoodPandaTest.java (hunk 5 above)
Comment (TatsianaKazachenka) [reply src/test/java/tests/MoodPandaTest.java:15]: done

**Pull request merged**